pkg/services/provisioning/apps/app_provisioner_test.go

```diff
@@ -0,0 +1,95 @@
+package apps
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAppProvisioner(t *testing.T) {
+	t.Run("Should return error when config reader returns error", func(t *testing.T) {
+		expectedErr := errors.New("test")
+		reader := &testConfigReader{err: expectedErr}
+		ap := AppProvisioner{log: log.New("test"), cfgProvider: reader}
+		err := ap.applyChanges("")
+		require.Equal(t, expectedErr, err)
+	})
+
+	t.Run("Should apply configurations", func(t *testing.T) {
+		bus.AddHandler("test", func(query *models.GetOrgByNameQuery) error {
```

Comment:
I really think it's best to remove these handlers after the test, via `t.Cleanup`, so they won't be accidentally used by other tests.